target_snowflake/file_formats/file_format.py

```diff
@@ -0,0 +1,15 @@
+"""
+S3 Upload Client
+"""
+
+
+# pylint: disable=too-few-public-methods
+class FileFormat:
```

Comment:
What is the reasoning behind making this a class? do you forsee it growing to have a constructor and methods with some file format logic?

Reply:
At the moment it's only for a possible:
```
if '.txt' in
theoretically it could be a base class of any supported file format and related methods. Like `singer_records_to_file`

Reply:
At the moment it's only for a possible:
```
if '.txt' in
theoretically it could be a base class of any supported file format and related methods. Like `singer_records_to_file`

Reply:
At the moment it's only for a possible:
```
if '.txt' in
theoretically it could be a base class of any supported file format and related methods. Like `singer_records_to_file`

Reply:
At the moment it's only for a possible:
```
if '.txt' in
theoretically it could be a base class of any supported file format and related methods. Like `singer_records_to_file`

Reply:
at the moment I have plans only to add PARQUET to do list and do something like this:

```
if file_format_type not in FileFormat.supported_types:
    raise InvalidFileFormatException(
        f"Not supported named file format {file_format_name}. Supported file formats: {FileFormat.file_formats}")
```

Well, that can be done differently, indeed. But we can also extend this little class later to be the base class of supported file types, like csv or parquet. There are common tasks that both file types should implement in the same or a little bit differently.

What do you think would that make sense, or not really?

Reply:
hmm ok, makes a bit of sense, I can see this class having little file format functionalities.

Reply:
please check bc338c20c03e721b6518956308cb23d18bbf5511. I converted it to a `FileFormatTypes` enum.

Reply:
Enums are great!!